subprojects/platform-play/src/main/java/org/gradle/play/internal/run/PlayWorkerClient.java

```diff
@@ -24,17 +24,21 @@
 import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.SynchronousQueue;
 
-public class PlayWorkerClient implements PlayRunWorkerClientProtocol, BuildGateToken.GateKeeper {
+public class PlayWorkerClient implements PlayRunWorkerClientProtocol {
     private static final Logger LOGGER = Logging.getLogger(PlayWorkerClient.class);
+    public static final String GATED_BUILD_SYSPROP = "org.gradle.internal.play.gated";
 
-    private final BuildGateToken gateToken;
+    private final BuildGateToken.GateKeeper gateKeeper;
     private final BlockingQueue<PlayAppStart> startEvent = new SynchronousQueue<PlayAppStart>();
     private final BlockingQueue<PlayAppStop> stopEvent = new SynchronousQueue<PlayAppStop>();
     private int gateCount;
 
     public PlayWorkerClient(BuildGateToken gateToken) {
-        this.gateToken = gateToken;
-        gateToken.addGateKeeper(this);
+        if (Boolean.getBoolean(GATED_BUILD_SYSPROP)) {
```

Comment:
Should default to `true` for now. Or perhaps rename the property so setting to 'true' means 'do not wait for browser before building'. `org.gradle.internal.play.eagerRebuild`?